projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,327 @@
+        String content = contentBuilder.toString();
+        return content;
+    }
+
+    /**
+     * Fill lawsuit template with user & spam company's details.
+     * @return Template's content as Html, with fields replaced.
+     */
+    public String fillInTemplate() {
+
+        // Read template's content:
+        String res = templateContent;
+
+        // Fill in general form details:
+        claimCase = sentHaser ? claimCaseHaser: claimCaseSubscription;
+        fiveLawsuitsStatus = sentMoreThanFiveLawsuits ? moreThanFiveLawsuits: lessThanFiveLawsuits;
+        res = res.replace("&lt;claimCase&gt;", claimCase);
+        res = res.replace("&lt;claimAmount&gt;", claimAmount);
+        res = res.replace("&lt;spamType&gt;", spamType);
+        res = res.replace("&lt;moreThanFiveLawsuits&gt;", fiveLawsuitsStatus);
+        res = res.replace("&lt;undefined&gt;", "");
+        res = res.replace("&lt;receivedSpamDate&gt;", dateFormatter.format(receivedSpamDate));
+        res = res.replace("&lt;currentDate&gt;", dateFormatter.format(currentDate));
+
+        // User details:
+        res = res.replace("&lt;userFullName&gt;", userPrivateName + " " + userLastname);
+        res = res.replace("&lt;userId&gt;", userID);
+        res = res.replace("&lt;userAddress&gt;", userAddress);
+        res = res.replace("&lt;userPhone&gt;", userPhone);
+        res = res.replace("&lt;userFax&gt;", userFax);
+
+        // First company's details:
+        res = res.replace("&lt;companyName&gt;", companyName);
+        res = res.replace("&lt;companyId&gt;", companyId);
+        res = res.replace("&lt;companyAddress&gt;", companyAddress);
+        res = res.replace("&lt;companyPhone&gt;", companyPhone);
+        res = res.replace("&lt;companyFax&gt;", companyFax);
+
+        // Second company's details:
+        res = res.replace("&lt;company2Name&gt;", company2Name);
+        res = res.replace("&lt;company2Id&gt;", company2Id);
+        res = res.replace("&lt;company2Address&gt;", company2Address);
+        res = res.replace("&lt;company2Phone&gt;", company2Phone);
+        res = res.replace("&lt;company2Fax&gt;", company2Fax);
+
+        return res;
+    }
+
+    /**
+     * Genereates the lawsuit Pdf file name according to Time & Date user had confirmed the form.
+     */
+    public void createPdfLawsuitFileName(){
+
+        lawsuitOutputPdfFileName = dateAndTimeFormatter.format(new Date()) + ".pdf";
+        lawsuitOutputPdfFilePath = Paths.get(lawsuitOutputPath,lawsuitOutputPdfFileName).toString();
+    }
+
+    /**
+     * This is where configurations of folders structure of Lawsuit Acitivity is been done.
+     * In case where directories don't exists, it creates them.
+     */
+    public void initDirStructure(){
+
+        // Main LawsuitActivity path:
+        // app_name/lawsuit_main_dir/
+        lawsuitMainPath = Paths.get(Environment.getExternalStorageDirectory().getPath(),getString(R.string.app_name),LAWSUIT_MAIN_DIR_NAME).toString();
```

Comment:
Looks like `lawsuitMainPath` is not used. If you skip creating it, and create only `lawsuitTemplateDirPath`, will it create it recursively?

Reply:
I wanted to avoid creating them recursively, so it will be easier to change the structure "between" them.

Changed it. Left the lawsuitMainPath to be as beginning of all other paths.